pubsub/cloud-client/subscriber.py

```diff
@@ -171,6 +221,113 @@ def update_subscription(project_id, subscription_name, endpoint):
     # [END pubsub_update_push_configuration]
 
 
+def update_subscription_with_dead_letter_policy(
+    project_id, topic_name, subscription_name, dead_letter_topic_name
+):
+    """Update a subscription's dead letter policy."""
+    # [START pubsub_dead_letter_update_subscription]
+    from google.cloud import pubsub_v1
+    from google.cloud.pubsub_v1.types import DeadLetterPolicy
+
+    # TODO project_id = "Your Google Cloud Project ID"
+    # This is an existing topic that the subscription with dead letter policy
+    # is attached to.
+    # TODO topic_name = "Your Pub/Sub topic name"
+    # This is an existing subscription with a dead letter policy.
+    # TODO subscription_name = "Your Pub/Sub subscription name"
+    # This is an existing dead letter topic that the subscription with dead
+    # letter policy will forward dead letter messages to.
+    # TODO dead_letter_topic_name = "Your Pub/Sub dead letter topic name"
+
+    subscriber = pubsub_v1.SubscriberClient()
+    topic_path = subscriber.topic_path(project_id, topic_name)
+    subscription_path = subscriber.subscription_path(
+        project_id, subscription_name
+    )
+    dead_letter_topic_path = subscriber.topic_path(
+        project_id, dead_letter_topic_name
+    )
+
+    subscription_before_update = subscriber.get_subscription(subscription_path)
+    print("Before the update: {}".format(subscription_before_update))
```

Comment:
Can you add a comment explaining to the user what is happening here? e.g. "Get the current version of the subscription so we can update it."

Reply:
It would be a little misleading to phrase it like that. The print statement is mainly there so people can compare fields. To update a subscription, an expected subscription (not the original subscription) and a field mask are supplied to the update method.